src/config.rs

```diff
@@ -163,14 +159,14 @@ mod tests {
     fn parse_starknet_config() {
         let expected_seq_addr = contract_address!(SEQUENCER_ADDR_0_13_0);
 
-        let conf = StarknetGeneralConfig::default();
+        let conf = StarknetGeneralConfig::from_default_file().expect("Failed to load default config file");
 
         assert!(!conf.constant_gas_price);
         assert!(conf.enforce_l1_handler_fee);
 
-        assert_eq!(4000000, conf.invoke_tx_max_n_steps);
+        assert_eq!(1000000, conf.invoke_tx_max_n_steps);
```

Comment:
Where did that value change?

Reply:
It's defined two ways:
* in `cairo-lang` it comes from `general_config.yml` where it's set to `1_000_000`
* in `blockifier` as `pub const MAX_STEPS_PER_TX: usize = 4_000_000;`

I don't think these values have changed recently.

However, the previous behavior of `Default` for `StarknetGeneralConfig` was done in such a way that it tried to pull values from the file `./cairo-lang/src/starkware/starknet/definitions/general_config.yml`. If that failed (either because it didn't exist or threw a parse error), it would fall back to hard-coded values. In this case, `4000000` was the hard-coded value (`MAX_STEPS_PER_TX` in `blockifier`) and `1000000` comes from a successful parse of `cairo-lang`'s config file.

Essentially, this behavior was silently masking a parse error (`cairo_resource_fee_weights` was removed) and giving us static values instead.